src/main/java/org/opensearch/security/auditlog/sink/InternalOpenSearchSink.java

```diff
@@ -85,6 +86,7 @@ public boolean doStore(final AuditMessage msg) {
                     .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
                     .setSource(msg.getAsMap());
                 threadPool.getThreadContext().putHeader(ConfigConstants.OPENDISTRO_SECURITY_CONF_REQUEST_HEADER, "true");
+                irb.setOpType(DocWriteRequest.OpType.CREATE);
```

Comment:
Please add new suite of test cases for audit logs that uses datastream.  I would recommend extending the existing 'basic' test cases [1] and modify them slightly to set a datastream instead of the default index.

- [1] https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/auditlog/integration/BasicAuditlogTest.java